aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -0,0 +1,324 @@
+    int collapseIndicesDim = indicesInfo.collapseDims(dim);               \
+    int64_t inputSlices = 1;                                              \
+    for (int i = 0; i < inputInfo.dims; ++i) {                            \
+      inputSlices *= inputInfo.sizes[i];                                  \
+    }                                                                     \
+    int64_t topKSlices = 1;                                               \
+    for (int i = 0; i < topKInfo.dims; ++i) {                             \
+      topKSlices *= topKInfo.sizes[i];                                    \
+    }                                                                     \
+                                                                          \
+    dim3 grid;                                                            \
+    TORCH_INTERNAL_ASSERT(getGridFromTiles(inputSlices, grid), "Too many slices to sort"); \
+                                                                          \
+    dim3 block(std::min(at::cuda::ATenCeilDiv(sliceSize, (int64_t) C10_WARP_SIZE)*(int64_t) C10_WARP_SIZE, (int64_t) 1024)); \
+                                                                          \
+    /* This is used as a template parameter to calculate indices. */      \
+    /* We only specialize it if all collapsed dim sizes are the */        \
+    /* same; otherwise, we use -1 which is the specialization */          \
+    /* parameter for arbitrary dimensions */                              \
+    int allDims = inputInfo.dims;                                         \
+    if (topKInfo.dims != allDims || indicesInfo.dims != allDims) {        \
+      allDims = -1;                                                       \
+    }                                                                     \
+                                                                          \
+    RUN_DIM(INDEX_T);                                                     \
+  });
+
+  // the below is safe with 0-dimensional tensors because it is based on
+  // TensorInfo which implicitly expands to 1-dimensional.
+  if (input.numel() > 0) {
+    // Based on required index size, run the algorithm with the
+    // appropriate index type
+    if (at::cuda::detail::canUse32BitIndexMath(input) &&
+        at::cuda::detail::canUse32BitIndexMath(values) &&
+        at::cuda::detail::canUse32BitIndexMath(indices)) {
+      RUN_T(uint32_t);
+    } else {
+      RUN_T(uint64_t);
+    }
+  }
+#undef RUN_T
+#undef RUN_DIM
+#undef RUN_DIR
+#undef RUN_K
+
+  // Sort the results if the user wants them sorted, since our
+  // selection routine does not ensure sorting
+  if (sorted && values.numel() > 1) {
+    if (should_use_small_sort(values, dim)) {
+      // This avoids any memory allocations and performs all sorting
+      // work inplace along the slice
+
+      sortKeyValueInplace(values, indices, dim, largest);
```

Comment:
Same here, `values.numel() > 1` above guarantees that `sortKeyValueInplace` will not be called for 0D tensors. So no need to handle 0D tensor in `sortKeyValueInplace`.

Reply:
Yup, looks like it works (just going through some tests now to double-check).